src/index.ts

```diff
@@ -60,9 +63,32 @@ export async function downloadEntityAndContentFiles(
   }
 
   const entityMetadata: {
+    type: string,
+    metadata?: any,
     content?: Array<ContentMapping>
   } = JSON.parse(contentStream)
 
+  if (entityMetadata.type === 'profile' && entityMetadata.metadata) {
```

Comment:
can we add a comment here explaining why this is needed? maybe referencing the issue?

Reply:
Done